eng/build.sh

```diff
@@ -36,6 +36,7 @@ usage()
   echo "  --prepareMachine           Prepare machine for CI run, clean up processes after build"
   echo "  --warnAsError              Treat all warnings as errors"
   echo "  --sourceBuild              Simulate building for source-build"
+  echo "  --build-full-solution      Build Roslyn.sln instead of Compilers.sln"
```

Comment:
We should switch to powershell core so we can just re-use build.ps1 on Unix and not have to maintain two different scripts

Reply:
would be nice, but don't we still need something to bootstrap getting dotnet on the machine?

Reply:
I think we discussed this around the time we started running tests on helix. IIRC, unless arcade decides to take that plunge, it will be hard for us to do it, because the Unix specific behaviors won't be in the arcade powershell scripts.

Reply:
> We should switch to powershell core so we can just re-use build.ps1 on Unix and not have to maintain two different scripts

I would love for this to be the case, but arcade does not guarantee that their powershell scripts will correctly on non-Windows, so I don't think this is an option.

Reply:
I'm certainly not opposed to it. I just remember getting pushback the last time I asked for it.